### PR TITLE
Tech/validate branch name in pre push

### PR DESCRIPTION
**Pull request by alschmut**
# Tech/validate branch name in pre push

## Description

Because git sometimes makes problems when branch names use uppercase letters, here is an idea to check the branch name before pushing
- Abort `git push` if branch name does not match the branch naming convention in [CONTRIBUTING.md](https://github.com/MaibornWolff/codecharta/blob/master/CONTRIBUTING.md)
- It expects following structure `<branch-type>/[<issue-number>/]<branch-name>`
- It prints useful messages
  - to not use uppercase letters and
  - to use the dash to split words instead of the underscore character and
  - how to rename a branch using the CLI

## Definition of Done

A task/pull request will not be considered to be complete until all these items can be checked off.

- [x] **All** requirements mentioned in the issue are implemented
- [x] Does match the Code of Conduct and the Contribution file
- [x] Task has its own **GitHub issue** (something it is solving)
  - [x] Issue number is **included in the commit messages** for traceability
- [x] **Update the README.md** with any changes/additions made
- [x] **Update the CHANGELOG.md** with any changes/additions made
- [x] **Enough test coverage to ensure that uncovered changes do not break functionality**
- [x] **All tests pass**
- [x] **Descriptive pull request text**, answering:
  - What problem/issue are you fixing?
  - What does this PR implement and how?
- [x] **Assign your PR to someone** for a code review
  - This person _will be contacted **first**_ if a bug is introduced into `master`
- [x] **Manual testing** did not fail

### CODE REVIEW

**Comment by NearW** [comment]
I moved the sh-script to the script folder and removed the deprecated tslint rule for unused variables.

File: visualization/package.json [hunk 1]
```diff
@@ -124,7 +124,7 @@
 	"husky": {
 		"hooks": {
 			"pre-commit": "npm run format:quick",
-			"pre-push": "npm run lint"
+			"pre-push": "npm run lint && sh ./pre-push.sh"
```
[the remark this answers is not in the document]
Comment (alschmut) [reply visualization/package.json]: okey, on which machine are you working?

**Comment by alschmut** [comment]
> I moved the sh-script to the script folder and removed the deprecated tslint rule for unused variables.

Why did you remove the TSLint rule??

**Comment by NearW** [comment]
> Why did you remove the TSLint rule??

Because we always get the error message, that this rule is deprecated and the built in typescript compiler is taking care of it.

File: visualization/package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (NearW) [reply visualization/package.json]: mac os

**Comment by NearW** [comment]
TSLint rule re-enabled again. If we disable the rule, we have to activate `noUnusedLocals` in the `tsconfig` and use `@ts-ignore` on each line where we have unused injections

**Pull request merged**